### PR TITLE
Adding more options to the yum_repository module

### PR DESCRIPTION
**Pull request by jtyr**
This PR is a replacement of PR #1595. It helps to separate the shipment of the new parameters from adding of the Managed Mode which is under discussion in the PR #1728.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @jtyr. Since you are a maintainer of this module, we are marking this PR for inclusion.

File: packaging/os/yum_repository.py [hunk 1]
```diff
@@ -44,23 +51,41 @@
         (bytes/sec) then this option is ignored. Default is C(0) (no bandwidth
         throttling).
   baseurl:
-    required: false
+    required: true
```
Comment (resmo) [inline packaging/os/yum_repository.py]: you can keep it as required=false if you have conditional requirements as in the description, required: true should be identical to the params in argument_spec=dict()

Comment (jtyr) [inline packaging/os/yum_repository.py]: Thanks for pointing this out. It's fixed now.

**Comment by abadger** [comment]
Thanks!  merged to devel.

**Pull request merged**